Orange/widgets/utils/saveplot.py

```diff
@@ -0,0 +1,42 @@
+import os.path
+from operator import attrgetter
+
+from PyQt4 import QtGui, QtCore
+
+
+# noinspection PyBroadException
+def save_plot(data, file_formats, filename=""):
+    formats = [(f.DESCRIPTION, f.EXTENSIONS)
+               for f in sorted(set(file_formats.values()),
+                               key=attrgetter("OWSAVE_PRIORITY"))]
+    filters = ['{} (*{})'.format(desc, ' *'.join(exts))
+               for desc, exts in formats]
+
+    _LAST_DIR_KEY = "directories/last_graph_directory"
+    _LAST_EXT_KEY = "directories/last_graph_extension"
+    settings = QtCore.QSettings()
+    start_dir = settings.value(_LAST_DIR_KEY, filename)
+    if not start_dir or not os.path.exists(start_dir):
+        start_dir = os.path.expanduser("~")
+    last_ext = settings.value(_LAST_EXT_KEY, "")
+    if last_ext not in filters:
+        last_ext = filters[0]
+
+    filename, filter = QtGui.QFileDialog.getSaveFileNameAndFilter(
+        None, 'Save as ...', start_dir, ';;'.join(filters), last_ext)
+    if not filename:
+        return
+
+    ext = os.path.splitext(filename)[1]
+    exts = formats[filters.index(filter)][1]
```

Comment:
I now changed that in PR#946. Extension is added only if there is none. It can't happen on Os X, though -- the dialog always returns correct extension. See https://github.com/biolab/orange3/pull/946/files#diff-0a256bbfcab13a3efd498498377c9fa9R29.

These to two PRs are related and should be "squashed" together. (Should I? How? Just cherry pick commits from both branches into one?)